### PR TITLE
Load xdebug on demand for VSCode and Docker

### PR DESCRIPTION
**Pull request by rafaelzaleski**
## Changes proposed in this Pull Request:

I noticed that the dev container was extremely slow, especially on Linux. After some investigation, I figured out it was because xdebug was enabled for every request, even if we were not using it.

This PR adds two tasks in the vscode debug config file to enable xdebug only when the debugger is started and disable it when it is stopped. It also updates the xdebug version in the Dockerfile to version 3.

- Update xdebug version on dev docker container to version 3
- Update Dockerfile to leave xdebug disabled by default
- Update vscode config file to enable and disable xdebug on demand

## Testing instructions

Please test if the debug is running without additional configurations needed:

1. Run `npm run down` to stop your dev environment if it's running.
2. Run `docker-compose build --no-cache` to rebuild the development image
3. Run `npm run up` to start the containers.
4. Open the Visual Studio Code in the project root folder
5. Add a breakpoint in the main plugin file
6. Start the debugger
7. Visit the dev site on [http://localhost:8082/](http://localhost:8082/)
8. The debugger should stop in your breakpoint.
9. Stop the debugger.

If the debugger is working, everything is good. The speed gain is more noticeable on Linux. 

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [x] Tested on mobile (or does not apply)

**Post merge**

-   [x] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -8,11 +8,14 @@
             "name": "Listen for Xdebug (remote/docker)",
             "type": "php",
             "request": "launch",
-            "port": 9000,
+            "hostname": "0.0.0.0",
```
Comment (ricardo) [inline .vscode/launch.json:11]: Is it necessary to specify the hostname?

File: .vscode/launch.json [hunk 2]
```diff
@@ -8,11 +8,14 @@
...
+            "port": 9003,
```
Comment (ricardo) [inline .vscode/launch.json]: If we keep the same port, people don't necessarily need to rebuild their containers for debugging to continue to work.

**Review by ricardo — APPROVED** [review]
LGTM and tests well. Pre-approving 👍 

I think it's worth it to set `xdebug.client_port=9000` for compatibility with existing containers.

Can you give everyone a heads up in #woo-stripe (for Fractal, Axiom and everyone involved) about the change, in case someone uses a different editor, and to let people know they need to rebuild their docker containers in order for this to work?

File: .vscode/launch.json (hunk 2 above)
Comment (rafaelzaleski) [reply .vscode/launch.json]: Good point

File: .vscode/launch.json (hunk 1 above)
Comment (rafaelzaleski) [reply .vscode/launch.json:11]: Yes, it doesn't connect back if we don't.

**Comment by rafaelzaleski** [comment]
> LGTM and tests well. Pre-approving +1
> 
> I think it's worth it to set `xdebug.client_port=9000` for compatibility with existing containers.

Done

> Can you give everyone a heads up in #woo-stripe (for Fractal, Axiom and everyone involved) about the change, in case someone uses a different editor, and to let people know they need to rebuild their docker containers in order for this to work?

I'll do that. Since people can be using another IDE, I moved the command to start and stop xdebug to an npm command.

**Pull request merged**